subsys/shell/modules/kernel_service.c

```diff
@@ -220,6 +220,9 @@ static int cmd_kernel_reboot_warm(const struct shell *shell,
 {
 	ARG_UNUSED(argc);
 	ARG_UNUSED(argv);
+#if (CONFIG_KERNEL_SHELL_REBOOT_DELAY > 0)
```

Comment:
To be considered if `IS_ENABLED` is better. @nordic-krch  any thoughts?

Reply:
`IS_ENABLED` cannot be used because it is not binary value you might do
```
if (CONFIG_KERNEL_SHELL_REBOOT_DELAY > 0) {
   k_sleep();
}
```
but compilation will fail when CONFIG_REBOOT is not defined. So i guess it should stay as is.